qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -404,6 +404,46 @@ def simplify(self, atol=None, rtol=None):
             x = self.paulis.x[non_zero_indexes]
             z = self.paulis.z[non_zero_indexes]
             coeffs = coeffs[non_zero]
+
+        return SparsePauliOp(
+            PauliList.from_symplectic(z, x), coeffs, ignore_pauli_phase=True, copy=False
+        )
+
+    def chop(self, tol=1e-14):
+        """Remove real and imaginary parts of the coefficients that are close to 0.
+
+        For example, the operator representing ``1+1e-17j X + 1e-17 Y`` with a tolerance larger
+        than ``1e-17`` will be reduced to ``1 X`` whereas :meth:`.SparsePauliOp.simplify` would
+        return ``1+1e-17j X``.
+
+        Args:
+            tol (float): The absolute tolerance to check whether a real or imaginary part is 0.
+
+        Returns:
+            SparsePauliOp: This operator with chopped coefficients.
+        """
+        # Remove real (resp. imaginary) parts that are close to 0. This part for instance
+        # truncates 1+1e-17j to 1.0. Note that all coefficients that are left at this point
+        # are guaranteed to have a value above tolerance due to the previous is_zero check.
+        realpart_nonzero = np.logical_not(np.isclose(self.coeffs.real, 0, atol=tol))
+        imagpart_nonzero = np.logical_not(np.isclose(self.coeffs.imag, 0, atol=tol))
```

Comment:
This is probably a little more efficient as `realpart_nonzero = np.abs(self.coeffs.real) > tol`